docs/spark/how-to-guides/ubuntu-instructions.md

```diff
@@ -0,0 +1,231 @@
+
+     - Run the following command:
+
+       ```bash
+       mkdir -p ~/bin/maven
+       cd ~/bin/maven
+       wget https://www-us.apache.org/dist/maven/maven-3/3.6.0/binaries/apache-maven-3.6.0-bin.tar.gz
+       tar -xvzf apache-maven-3.6.0-bin.tar.gz
+       ln -s apache-maven-3.6.0 current
+       export M2_HOME=~/bin/maven/current
+       export PATH=${M2_HOME}/bin:${PATH}
+       source ~/.bashrc
+       ```
+       
+       Note that these environment variables will be lost when you close your terminal. If you want the changes to be permanent, add the `export` lines to your `~/.bashrc` file.
+
+     - Verify you are able to run `mvn` from your command-line       
+       Sample mvn -version output:
+       
+       ```
+       Apache Maven 3.6.0 (97c98ec64a1fdfee7767ce5ffb20918da4f719f3; 2018-10-24T18:41:47Z)
+       Maven home: ~/bin/apache-maven-3.6.0
+       Java version: 1.8.0_191, vendor: Oracle Corporation, runtime: /usr/lib/jvm/java-8-openjdk-amd64/jre
+       Default locale: en, platform encoding: UTF-8
+       OS name: "linux", version: "4.4.0-17763-microsoft", arch: "amd64", family: "unix"
+       ```
+
+  4. Install **[Apache Spark 2.3+](https://spark.apache.org/downloads.html)**.
+
+     - Download [Apache Spark 2.3+](https://spark.apache.org/downloads.html) and extract it into a local folder (e.g., `~/bin/spark-2.3.2-bin-hadoop2.7`)
+     - Add the necessary [environment variables](https://www.java.com/en/download/help/path.xml) `SPARK_HOME` e.g., `~/bin/spark-2.3.2-bin-hadoop2.7/`
+
+       ```bash
+       export SPARK_HOME=~/bin/spark-2.3.2-hadoop2.7
+       export PATH="$SPARK_HOME/bin:$PATH"
+       source ~/.bashrc
+       ```
+       
+       Note that these environment variables will be lost when you close your terminal. If you want the changes to be permanent, add the `export` lines to your `~/.bashrc` file.
+
+     - Verify you are able to run `spark-shell` from your command-line        
+        Sample console output:
+        
+        ```
+        Welcome to
+              ____              __
+             / __/__  ___ _____/ /__
+            _\ \/ _ \/ _ `/ __/  '_/
+           /___/ .__/\_,_/_/ /_/\_\   version 2.3.2
+              /_/
+
+        Using Scala version 2.11.8 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_201)
+        Type in expressions to have them evaluated.
+        Type :help for more information.
+
+        scala> sc
+        res0: org.apache.spark.SparkContext = org.apache.spark.SparkContext@6eaa6b0c
+        ```                      
+
+Make sure you are able to run `dotnet`, `java`, `mvn`, `spark-shell` from your command-line before you move to the next section. Feel there is a better way? Please [open an issue](https://github.com/dotnet/spark/issues) and feel free to contribute.
+
+## Build
+
+For the rest of the section, it is assumed that you have cloned the Spark .NET repository into your machine e.g., `~/dotnet.spark/`
```

Comment:
"It is assumed" might sound intimidating to new users (i.e. "if we're making assumptions, what else have I missed? Is the command below enough to clone the repo or are there other cloning steps I'm assumed to have done?").

```suggestion
For the remainder of this guide, you will need to have cloned the .NET for Apache Spark repository into your machine e.g., `~/dotnet.spark/`.
```